dask/delayed.py

```diff
@@ -219,20 +221,25 @@ def unpack_collections(expr, _return_collections=True):
         return args, collections
 
     if typ is dict:
-        args, collections = unpack_collections(
-            [[k, v] for k, v in expr.items()], _return_collections=False
+        keyargs, kcollections = unpack_collections(
+            [k for k in expr.keys()], _return_collections=False
         )
-        if not collections:
+        valargs, valcollections = unpack_collections(
+            [v for v in expr.values()], _return_collections=False
+        )
+        collections = kcollections + valcollections
+        args = Dict({k: v for k, v in zip(keyargs, valargs)})
```

Comment:
This is the bit that's failing for unhashable `keyargs` (like `Alias`). Seems like defining `Alias.__hash__` as something like `hash(self.key)` might work, but I'm not sure what other consequences that might have and what other types might end up here.

Reply:
I rewrote it to avoid the dict / hashing of the key entirely.